### DISCUSSION TITLE
Why not using the class member (weight, bias)?

### DISCUSSION THREAD

**Discussion by whamza15** [dmlc/gluon-nlp#340] [Q&A] [answered]
https://github.com/dmlc/gluon-nlp/blob/fb270337d3c0bbd0b74693eda3cda7d98ec81161/gluonnlp/model/sampled_block.py#L169

Why pass them from outside?

**Comment by szha** [comment]
The convention for Gluon is that the class members for parameters will be automatically passed in hybrid_forward as keyword arguments. The **name** of the keyword argument is from the self.params.get(**name**).

**Comment by whamza15** [comment]
Hmm, didn't know that. Not clear from the docs. Thanks for answering

**Comment by szha** [comment, marked as answer]
Thanks for the feedback. I created an issue in mxnet repo to track this apache/incubator-mxnet#12580
